### PR TITLE
UG: Add numbering

### PR DESCRIPTION
**Pull request by samuelim01**
It's slightly unintuitive to use the user guide with no numbering to the sections.

Let's add numbering to each section and cluster similar commands to make the user guide easier to use.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -11,84 +11,84 @@
...
-- [Appendix A: Acceptable values for parameters](#appendix-a-acceptable-values-for-parameters)
-- [Appendix B: Miscellaneous error messages](#appendix-b-miscellaneous-error-messages)
+1. [Quick Start](#1-quick-start)
+2. [Features](#2-features)<br>
+    2.1. [Add Command](#21-add-command)<br>
+    &emsp; 2.1.1. [Adding a guest: `guest add`](#211-adding-a-guest--guest-add)<br>
+    &emsp; 2.1.2. [Adding a vendor: `vendor add`](#212-adding-a-vendor--vendor-add)<br>
+    2.2. [Delete Command](#22-delete-command)<br>
+    &emsp; 2.2.1. [Deleting a guest: `guest delete`](#221-deleting-a-guest--guest-delete)<br>
+    &emsp; 2.2.1. [Deleting a vendor: `vendor delete`](#222-deleting-a-vendor--vendor-delete)<br>
```
Comment (keaganpzh) [inline docs/UserGuide.md:21]: this should be 2.2.2

File: docs/UserGuide.md [hunk 2]
```diff
@@ -11,84 +11,84 @@
...
+    2.3. [Edit Command](#23-edit-command)<br>
+    &emsp; 2.3.1. [Editing a guest: `guest edit`](#231-editing-a-guest--guest-edit)<br>
+    &emsp; 2.3.2. [Editing a vendor: `vendor edit`](#232-editing-a-vendor--vendor-edit)<br>
+    2.4. [List Command](#24-list-command)<br>
+    &emsp; 2.4.1. [Viewing all guests: `guest list`](#241-viewing-all-guests--guest-list)<br>
+    &emsp; 2.4.2. [Viewing all vendors: `vendor list`](#242-viewing-all-vendors--vendor-list)<br>
+    2.5. [Filter Command](#25-filter-command)<br>
+    &emsp; 2.5.1. [Filtering guests: `guest filter`](#251-filtering-guests--guest-filter)<br>
+    &emsp; 2.5.2. [Filtering vendor: `vendor filter`](#252-filtering-vendors--vendor-filter)<br>
+    2.6. [General Commands](#26-general-commands)<br>
```
Comment (keaganpzh) [inline docs/UserGuide.md:31]: I'm wondering if the section on general commands should come before the rest of the commands. Do you think that will be better?

**Review by keaganpzh — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 2 above)
Comment (samuelim01) [reply docs/UserGuide.md:31]: I'm not too sure on this point, if it was just help, I would agree with you, but I'm not entirely convinced that users want to see all 4 general commands such as undo before the rest of the commands. Perhaps we can put this up for discussion?

File: docs/UserGuide.md (hunk 1 above)
Comment (samuelim01) [reply docs/UserGuide.md:21]: Good catch, thanks!

File: docs/UserGuide.md (hunk 2 above)
Comment (samuelim01) [reply docs/UserGuide.md:31]: Resolved as majority voted for section on general commands to come after the rest of the commands.

**Review by keaganpzh — APPROVED** [review]
LGTM

**Pull request merged**